server/models/campaign.py

```diff
@@ -166,6 +167,18 @@ def as_dict(self):
             self, recurse=False, exclude=[Note.room, Note.location, Note.user]
         )
 
+class Marker(BaseModel):
+    from .shape import Shape
+
+    uuid = ForeignKeyField(Shape, backref="markers", on_delete="CASCADE")
+    user = ForeignKeyField(User, backref="markers", on_delete="CASCADE")
+    location = ForeignKeyField(Location, backref="markers", on_delete="CASCADE")
+
+    def __repr__(self):
+        return f"<Marker {self.shape.uuid} {self.location.get_path()} - {self.user.name}"
```

Comment:
self.shape does not exist.  to fix it it should be self.uuid.uuid, which is kinda awkward.  So I suggest that you rename the uuid field to shape (don't forget to change the save.py migration query and marker.py socket file if you do this)

Reply:
Changed.